quill-sql/src/main/scala/io/getquill/generic/GenericDecoder.scala

```diff
@@ -341,22 +381,50 @@ object ConstructDecoded {
     val tpe = TypeRepr.of[T]
     val constructor = TypeRepr.of[T].typeSymbol.primaryConstructor
     // If we are a tuple, we can easily construct it
-    if (tpe <:< TypeRepr.of[Tuple]) {
-      val construct =
-        Apply(
-          TypeApply(
-            Select(New(TypeTree.of[T]), constructor),
-            types.map { tpe =>
-              tpe match {
-                case '[tt] => TypeTree.of[tt]
-              }
-            }
-          ),
-          terms.map(_.asTerm)
-        )
-      // println(s"=========== Create from Tuple Constructor ${Format.Expr(construct.asExprOf[T])} ===========")
-      construct.asExprOf[T]
-      // If we are a case class with no generic parameters, we can easily construct it
+    if (tpe <:< TypeRepr.of[EmptyTuple]) {
+      '{EmptyTuple}
+    } else if (tpe <:< TypeRepr.of[Tuple]) {
+      '{scala.runtime.Tuples.fromIArray(IArray(${Varargs(terms)})).asInstanceOf[T]}
+      // Alternative version:
```

Comment:
Why keeping commented code? 🤔

Reply:
I wanted to consult which version is preferred - array based or recursion-based.